vms/platformvm/txs/fee/calculator.go

```diff
@@ -0,0 +1,142 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package fee
+
+import (
+	"time"
+
+	"github.com/ava-labs/avalanchego/utils/constants"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs"
+	"github.com/ava-labs/avalanchego/vms/platformvm/upgrade"
+)
+
+var _ txs.Visitor = (*calculator)(nil)
+
+func NewStaticCalculator(config StaticConfig, upgradeTimes upgrade.Config) Calculator {
+	return Calculator{
+		config:       config,
+		upgradeTimes: upgradeTimes,
+	}
+}
+
+type Calculator struct {
+	config       StaticConfig
+	upgradeTimes upgrade.Config
+}
+
+// [CalculateFee] returns the minimal fee needed to accept [tx], at chain time [time]
+func (c Calculator) CalculateFee(tx txs.UnsignedTx, time time.Time) uint64 {
+	tmp := &calculator{
+		upgrades:  c.upgradeTimes,
+		staticCfg: c.config,
+		time:      time,
+	}
+
+	// this is guaranteed to never return an error
+	_ = tx.Visit(tmp)
+	return tmp.fee
+}
```

Comment:
Should we be passing around a `*Calculator` rather than a `Calculator`? This struct is pretty large to be passing by value no?

Reply:
Done